tests/test_cases/test_cocotb/test_queues.py

```diff
@@ -256,3 +256,88 @@ async def test_str_and_repr(_):
     s = repr(q)
     assert "_getters" not in s
     assert str(q)[:-1] in s
+
+
+@cocotb.test()
+async def test_waits(_):
```

Comment:
It's kind of a long tests. Maybe you can split it in 2: 1 for full, 1 for empty?